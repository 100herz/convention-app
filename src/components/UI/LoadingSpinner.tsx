import React from 'react'
import { ActivityIndicator, ActivityIndicatorProps, StyleSheet, View } from 'react-native'

import { colors, defaultStyles, DefaultStyles } from '@styles/theme'

const LoadingSpinner: React.FC<ActivityIndicatorProps> = ({ color = colors.accentColor, size = 'large', ...rest }) => {
  return (
    <View style={styles.container}>
      <ActivityIndicator {...rest} color={color} size={size} />
    </View>
  )
}

const styles = StyleSheet.create<DefaultStyles>({
  ...defaultStyles,
  container: {
    ...defaultStyles.container,
    justifyContent: 'center',
  },
})

export default LoadingSpinner
