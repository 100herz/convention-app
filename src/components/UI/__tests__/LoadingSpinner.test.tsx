import React from 'react'
import { render } from '@testing-library/react-native'

import LoadingSpinner from '../LoadingSpinner'

describe('<LoadingSpinner />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<LoadingSpinner />)
    expect(toJSON()).toMatchSnapshot()
  })
})
