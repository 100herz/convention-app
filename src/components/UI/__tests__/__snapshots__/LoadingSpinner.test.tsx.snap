// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#872232"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
